pkg/generator/generator.go

```diff
@@ -67,7 +67,7 @@ func (k *installGenerator) GenerateInstallConfig(ctx context.Context, cluster co
 	log := logutil.FromContext(ctx, k.log)
 
 	entries, err := os.ReadDir(k.workDir)
-	if err == nil && len(entries) > 0 {
+	if k.Config.InstallInvoker == "agent-installer" && err == nil && len(entries) > 0 {
```

Comment:
As mentioned in my previous [comment](https://github.com/openshift/assisted-service/pull/7472#discussion_r2046587136) `k.workDir` is not the right directory to check to verify if the certs were already created for a given cluster, but the check should be performed against `clusterWorkDir`, since the `k.workDir` just contains the installer binaries.
Also, as in the previous case, it does not seem effectively required to additionally constraint the code on the agent-specific case

Reply:
As noticed by @bfournie `clusterWorkDir` points to a temp folder, but I guess it gets renamed to `k.workDir + cluster.ID.String()`, so the check should be performed against that folder name